### PR TITLE
Add proposal process

### PR DESCRIPTION
**Pull request by nikimanoledaki**
#### What type of PR is this?

`kind/documentation`

#### What this PR does / why we need it:

* Creates a proposal process through this repo
* Changes to proposals / design docs can be tracked through git
* Adds template with instructions (inspired by KEP/ADRs)
* Helps to deprecate the design doc (which doesn't scale well)

#### Which issue(s) this PR fixes:

https://github.com/cncf-tags/green-reviews-tooling/issues/47

#### Special notes for your reviewer (optional):

Please let me know what you think about creating a proposal process through this repository. It's similar to the KEP process in Kubernetes or ADRs which is widely used in OSS. Going through git would allow a better tracking of the history of proposals and how they change. It would also help to have discussions on individual points via PRs.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -37,3 +37,15 @@ Recommendations for a faster Pull Request review:
   - **Signed**: Include a `Signed-off-by: Author Name <authoremail@example.com>` in all commits by doing `git rebase HEAD~2 --signoff` (replace `~2` with the number of commits to sign) and then `git push -f`. More info [here](https://github.com/cncf-tags/green-reviews-tooling/pull/53/checks?check_run_id=21563565653).
   - **Verified**: [Learn more about commit signature verification with GPG.](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification)
 - The Kubernetes Best Practices for faster Reviews is a great resource for PR best practices: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
+
+### Proposals
+
+For larger feature requests, please submit a design proposal in [docs/proposals/](./docs/proposals/). This is similar to a [Kubernetes Enhancement Proposal (KEP)](https://github.com/kubernetes/enhancements) or a [Architecture Desicision Record (ADR)](https://github.com/joelparkerhenderson/architecture-decision-record).
```
Comment (rossf7) [inline CONTRIBUTING.md:43]:
```suggestion
For larger feature requests, please submit a design proposal in [docs/proposals/](./docs/proposals/). This is similar to a [Kubernetes Enhancement Proposal (KEP)](https://github.com/kubernetes/enhancements) or a [Architecture Decision Record (ADR)](https://github.com/joelparkerhenderson/architecture-decision-record).
```

File: docs/proposals/proposal-000-template.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+## Status
+
+<!--
+Must be one of provisional, implementable, implemented, deferred,
+rejected, withdrawn, or replaced.
+-->
+
+
+<!--
+The headings here are just starting points, add more as makes sense in what you
```
Comment (rossf7) [inline docs/proposals/proposal-000-template.md:36]:
```suggestion
The headings here are just starting points, add more as makes sense for what you
```

**Review by rossf7 — APPROVED** [review]
@nikimanoledaki I like this approach a lot and I think it can help a lot with larger features / deliverables the WG will work on.

I have more experience with ADRs than the KEP process. Looks like this is a lighter version of the KEP process?

It would be good to hear opinions from others who are more familiar with KEPs.

**Comment by dipankardas011** [comment]
Hi

**Comment by leonardpahlke** [comment]
Having some kind of design document with the usual questions is a good change. The KEP template is quite extensive. We will see which sections will make sense once we use it a few times. Lets try this out and make adjustments if needed.

**Pull request merged**